### PR TITLE
[stable/3.0] Provide .openrc for nodes with neutron-network role

### PR DESCRIPTION
**Pull request by jsuchome**
So we can execute neutron-ha-tool or nova commands if they
are in the different cluster than keystone.

OK, I'm not sure if this is correct, so it's rather initial proposal...

### CODE REVIEW

**Comment by jsuchome** [comment]
@dirkmueller this proposal is meant for Cloud6...

**Comment by jsuchome** [comment]
[(8) S54P6 (S49P16) Orchestrate upgrade of controller nodes in HA setup (upgrade step 12 part 2 -- HA/multiple clusters)](https://trello.com/c/MquyXrTn/252-8-s54p6-s49p16-orchestrate-upgrade-of-controller-nodes-in-ha-setup-upgrade-step-12-part-2-ha-multiple-clusters)

File: chef/cookbooks/keystone/recipes/openrc.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# useful with .openrc
+package "python-openstackclient"
+
+template "/root/.openrc" do
+  source "openrc.erb"
+  owner "root"
+  group "root"
+  mode 0600
+  variables(
+    keystone_settings: KeystoneHelper.keystone_settings(node, @cookbook_name)
```
Comment (vuntz) [inline chef/cookbooks/keystone/recipes/openrc.rb]: I'm afraid that the `@cookbook_name` bit here will not result in what you expect when included from neutron/nova recipes on nodes that are not keystone nodes...

Comment (jsuchome) [inline chef/cookbooks/keystone/recipes/openrc.rb]: Uhm... so you think I should rather use the template directly from `network_agents.rb` with `keystone_settings` retrieved from the neutron cookbook?

Comment (vuntz) [inline chef/cookbooks/keystone/recipes/openrc.rb]: Or you make this a LWRP where the attributes would be the node and a cookbook name?

Comment (jsuchome) [inline chef/cookbooks/keystone/recipes/openrc.rb]: Actually, are you sure there's something broken with this? The only part where `keystone_settings` uses `cookbook_name` is for setting `service_user`/`service_password` values and those are not needed by `openrc.erb`

Comment (vuntz) [inline chef/cookbooks/keystone/recipes/openrc.rb]: Isn't it also used to find the keystone server with the search query?

Comment (jsuchome) [inline chef/cookbooks/keystone/recipes/openrc.rb]: The search is there, but `cookbook` seems irrelevant https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/keystone/libraries/helpers.rb#L104

Comment (vuntz) [inline chef/cookbooks/keystone/recipes/openrc.rb]: @jsuchome it's used there https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/keystone/libraries/helpers.rb#L36 and this has an impact on the search query

**Comment by jsuchome** [comment]
@vuntz do you agree with this version of PR?
